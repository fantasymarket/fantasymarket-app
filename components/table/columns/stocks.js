import React from 'react';

import Link from 'next/link';
import formatMoney from 'utils/format-money';

export const columns = [
	{
		Header: 'Name',
		// eslint-disable-next-line react/prop-types
		accessor: ({ symbol, name }) => (
			<Link href="/stock/[symbol]" as={`/stock/${symbol}`}>
				<a>{name}</a>
			</Link>
		),
	},
	{
		Header: 'Symbol',
		accessor: 'symbol',
	},
	{
		Header: 'Price',
		accessor: 'price',
		Cell: ({ cell: { value } }) => {
			return formatMoney(parseFloat(value, 10));
		},
	},
	{
		Header: 'Market Cap',
		accessor: ({ price, shares }) => {
			return formatMoney(parseFloat(price, 10) * shares);
		},
	},
];

export const sampleData = [
	{
		name: 'Alphabet Inc.',
		symbol: 'GOOG',
		price: '10030',
		price24h: '9300',
		shares: 10000000,
	},
	{
		name: 'Apple Inc.		',
		symbol: 'APL',
		price: '25430',
		price24h: '23000',
		shares: 10000000,
	},
	{
		name: 'Microsoft',
		symbol: 'MSFT',
		price: '23432',
		price24h: '24342',
		shares: 10000000,
	},
	{
		name: 'ExxonMobil',
		symbol: 'EXN',
		price: '23342',
		price24h: '21342',
		shares: 10000000,
	},
];
