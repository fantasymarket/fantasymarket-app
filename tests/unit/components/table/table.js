import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Table from '@components/table';
import { columns, sampleData } from '@components/table/columns/stocks';

describe('Table component', () => {
	it('renders the component', async () => {
		const container = render(<Table columns={columns} data={sampleData} />);

		expect(container.baseElement).toMatchSnapshot();
	});

	it('can sort the table', async () => {
		const container = render(<Table columns={columns} data={sampleData} />);

		expect(container.baseElement).toMatchSnapshot();

		fireEvent(
			container.getByText('Name'),
			new MouseEvent('click', {
				bubbles: true,
				cancelable: true,
			}),
		);

		expect(container.baseElement).toMatchSnapshot();
	});
});
