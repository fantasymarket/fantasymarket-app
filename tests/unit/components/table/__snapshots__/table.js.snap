// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component can sort the table 1`] = `
<body>
  <div>
    <table
      class="sc-AxjAm dYOqdl"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Name
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Symbol
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Price
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Market Cap
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/GOOG"
            >
              Alphabet Inc.
            </a>
          </td>
          <td
            role="cell"
          >
            GOOG
          </td>
          <td
            role="cell"
          >
            $100.30
          </td>
          <td
            role="cell"
          >
            $1,003,000,000.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/APL"
            >
              Apple Inc.		
            </a>
          </td>
          <td
            role="cell"
          >
            APL
          </td>
          <td
            role="cell"
          >
            $254.30
          </td>
          <td
            role="cell"
          >
            $2,543,000,000.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/MSFT"
            >
              Microsoft
            </a>
          </td>
          <td
            role="cell"
          >
            MSFT
          </td>
          <td
            role="cell"
          >
            $234.32
          </td>
          <td
            role="cell"
          >
            $2,343,200,000.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/EXN"
            >
              ExxonMobil
            </a>
          </td>
          <td
            role="cell"
          >
            EXN
          </td>
          <td
            role="cell"
          >
            $233.42
          </td>
          <td
            role="cell"
          >
            $2,334,200,000.00
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;

exports[`Table component can sort the table 2`] = `
<body>
  <div>
    <table
      class="sc-AxjAm dYOqdl"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Name
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Symbol
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Price
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Market Cap
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/GOOG"
            >
              Alphabet Inc.
            </a>
          </td>
          <td
            role="cell"
          >
            GOOG
          </td>
          <td
            role="cell"
          >
            $100.30
          </td>
          <td
            role="cell"
          >
            $1,003,000,000.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/APL"
            >
              Apple Inc.		
            </a>
          </td>
          <td
            role="cell"
          >
            APL
          </td>
          <td
            role="cell"
          >
            $254.30
          </td>
          <td
            role="cell"
          >
            $2,543,000,000.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/MSFT"
            >
              Microsoft
            </a>
          </td>
          <td
            role="cell"
          >
            MSFT
          </td>
          <td
            role="cell"
          >
            $234.32
          </td>
          <td
            role="cell"
          >
            $2,343,200,000.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/EXN"
            >
              ExxonMobil
            </a>
          </td>
          <td
            role="cell"
          >
            EXN
          </td>
          <td
            role="cell"
          >
            $233.42
          </td>
          <td
            role="cell"
          >
            $2,334,200,000.00
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;

exports[`Table component renders the component 1`] = `
<body>
  <div>
    <table
      class="sc-AxjAm dYOqdl"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Name
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Symbol
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Price
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Market Cap
            <span>
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="18 15 12 9 6 15"
                />
              </svg>
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/GOOG"
            >
              Alphabet Inc.
            </a>
          </td>
          <td
            role="cell"
          >
            GOOG
          </td>
          <td
            role="cell"
          >
            $100.30
          </td>
          <td
            role="cell"
          >
            $1,003,000,000.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/APL"
            >
              Apple Inc.		
            </a>
          </td>
          <td
            role="cell"
          >
            APL
          </td>
          <td
            role="cell"
          >
            $254.30
          </td>
          <td
            role="cell"
          >
            $2,543,000,000.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/MSFT"
            >
              Microsoft
            </a>
          </td>
          <td
            role="cell"
          >
            MSFT
          </td>
          <td
            role="cell"
          >
            $234.32
          </td>
          <td
            role="cell"
          >
            $2,343,200,000.00
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <a
              href="/stock/EXN"
            >
              ExxonMobil
            </a>
          </td>
          <td
            role="cell"
          >
            EXN
          </td>
          <td
            role="cell"
          >
            $233.42
          </td>
          <td
            role="cell"
          >
            $2,334,200,000.00
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;
